import_export_celery/tasks.py

```diff
@@ -188,44 +188,58 @@ def run_import_job(pk, dry_run=True):
         return
 
 
-@task(bind=False)
+@task(bind=False,soft_time_limit=36000)
 def run_export_job(pk):
     log.info("Exporting %s" % pk)
     export_job = models.ExportJob.objects.get(pk=pk)
+    pks = json.loads(export_job.queryset)
+    qs_len = len(pks)
     resource_class = export_job.get_resource_class()
-    queryset = export_job.get_queryset()
-    qs_len = len(queryset)
+    pk_list =[],
+    rows = 0
+    interval = 100000
+
+    def TempFileSave(filename, content):
```

Comment:
Use snake_case for function names: https://en.wikipedia.org/wiki/Snake_case

Have you tried this append approach with formats other than csv? It seems that it would only work with appendable file formats.

Reply:
not yet, As per my knowledge, other formats will not support millions of rows